packages/rich-text/src/test/change-list-type.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+
+import { changeListType } from '../change-list-type';
+import { getSparseArrayLength } from './helpers';
+import { LINE_SEPARATOR } from '../special-characters';
+
+describe( 'changeListType', () => {
+	const ul = { type: 'ul' };
+	const ol = { type: 'ol' };
+
+	it( 'should only change list type if list item is indented', () => {
+		const record = {
+			formats: [ , ],
+			text: '1',
+			start: 1,
+			end: 1,
+		};
+		const result = changeListType( deepFreeze( record ), ul );
+
+		expect( result ).toEqual( record );
+		expect( result ).toBe( record );
+		expect( getSparseArrayLength( result.formats ) ).toBe( 0 );
+	} );
+
+	it( 'should change list type', () => {
+		const record = {
+			formats: [ , [ ul ] ],
+			text: `1${ LINE_SEPARATOR }`,
+			start: 2,
+			end: 2,
+		};
+		const expected = {
+			formats: [ , [ ol ] ],
+			text: `1${ LINE_SEPARATOR }`,
+			start: 2,
+			end: 2,
+		};
+		const result = changeListType( deepFreeze( record ), ol );
+
+		expect( result ).toEqual( expected );
+		expect( result ).not.toBe( record );
+		expect( getSparseArrayLength( result.formats ) ).toBe( 1 );
+	} );
+
+	it( 'should outdent with multiple lines selected', () => {
+		const record = {
+			formats: [ , [ ul ], , [ ul ], , [ ul, ul ], , [ ul ], , [ ul ], , , , [ ul ], , ],
+			text: `a${ LINE_SEPARATOR }1${ LINE_SEPARATOR }2${ LINE_SEPARATOR }i${ LINE_SEPARATOR }3${ LINE_SEPARATOR }4${ LINE_SEPARATOR }b`,
```

Comment:
These become really hard to read. I was going to ask if something like this would help:

```js
`
a
  ${ LINE_SEPARATOR }
  1
  ${ LINE_SEPARATOR }
  2
    ${ LINE_SEPARATOR }
    i
  ${ LINE_SEPARATOR }
  3
  ${ LINE_SEPARATOR }
  4
${ LINE_SEPARATOR }
b
`.replace(/[\n ]/g, '')
```

but once I'd written it I realised how awful it was. Also, I guess indentation isn't the point of this test, the point is for `text` to remain the same.

Perhaps just an inline comment mentioning that the reader isn't meant to follow the meaning of `text` here?

Reply:
Yeah, sounds good.